packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import Typography from '@mui/material/Typography';
+import { generateUtilityClasses } from '@mui/material';
+import { useTheme, styled } from '@mui/material/styles';
+import { useSlotProps } from '@mui/base/utils';
+import IconButton from '@mui/material/IconButton';
+import { ArrowLeft, ArrowRight } from '../icons';
+import { PickersArrowSwitcherProps } from './PickersArrowSwitcher.types';
+
+const classes = generateUtilityClasses('MuiPickersArrowSwitcher', ['root', 'spacer', 'button']);
+
+const PickersArrowSwitcherRoot = styled('div', {
+  name: 'MuiPickersArrowSwitcher',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{
+  ownerState: PickersArrowSwitcherProps;
+}>({
+  display: 'flex',
+});
+
+const PickersArrowSwitcherSpacer = styled('div', {
+  name: 'MuiPickersArrowSwitcher',
+  slot: 'Spacer',
+  overridesResolver: (props, styles) => styles.spacer,
+})<{
+  ownerState: PickersArrowSwitcherProps;
+}>(({ theme }) => ({
+  width: theme.spacing(3),
+}));
+
+const PickersArrowSwitcherButton = styled(IconButton, {
+  name: 'MuiPickersArrowSwitcher',
+  slot: 'Button',
+  overridesResolver: (props, styles) => styles.button,
+})<{
+  ownerState: PickersArrowSwitcherProps;
+}>(({ ownerState }) => ({
+  ...(ownerState.hidden && {
+    visibility: 'hidden',
+  }),
+}));
+
+export const PickersArrowSwitcher = React.forwardRef(function PickersArrowSwitcher(
+  props: PickersArrowSwitcherProps,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  const {
+    children,
+    className,
+    components = {},
+    componentsProps = {},
+    isLeftDisabled,
+    isLeftHidden,
+    isRightDisabled,
+    isRightHidden,
+    leftArrowButtonText,
+    onLeftClick,
+    onRightClick,
+    rightArrowButtonText,
+    ...other
+  } = props;
+  const theme = useTheme();
+  const isRtl = theme.direction === 'rtl';
+  const ownerState = props;
+
+  const LeftArrowButton = components.LeftArrowButton ?? PickersArrowSwitcherButton;
+  const leftArrowButtonProps = useSlotProps({
+    elementType: LeftArrowButton,
+    externalSlotProps: componentsProps.leftArrowButton,
+    additionalProps: {
+      size: 'small',
+      'aria-label': leftArrowButtonText,
+      title: leftArrowButtonText,
+      disabled: isLeftDisabled,
+      edge: 'end',
+      onClick: onLeftClick,
+    },
+    ownerState: { ...ownerState, hidden: isLeftHidden },
+    className: classes.button,
+  });
+
+  const LeftArrowIcon = components?.LeftArrowIcon ?? ArrowLeft;
+  const { ownerState: leftArrowIconOwnerState, ...leftArrowIconProps } = useSlotProps({
+    elementType: LeftArrowIcon,
+    externalSlotProps: componentsProps.leftArrowIcon,
+    ownerState: undefined,
+  });
+
+  const RightArrowButton = components.RightArrowButton ?? PickersArrowSwitcherButton;
+  const rightArrowButtonProps = useSlotProps({
+    elementType: LeftArrowButton,
+    externalSlotProps: componentsProps.leftArrowButton,
```

Comment:
```suggestion
    elementType: RightArrowButton,
    externalSlotProps: componentsProps.rightArrowButton,
```

Reply:
I was trying the following to avoid overflow in the clock view

```jsx
leftArrowButton: {
    color: "primary"
  },
  rightArrowButton: {
    color: "error"
  },
  ```
  
  
![image](https://user-images.githubusercontent.com/45398769/187668892-d0c812a1-d334-4efd-bfc7-f959a33233ec.png)

Reply:
Oups, fixed :p